infinite/index.ts

```diff
@@ -256,35 +252,46 @@ export const infinite = (<Data, Error>(useSWRNext: SWRHook) =>
       [infiniteKey, cache]
     )
 
-    // Function to load pages data from the cache based on the page size.
-    const resolvePagesFromCache = (pageSize: number): Data[] | undefined => {
-      // return an array of page data
-      const data: Data[] = []
-
-      let previousPageData = null
-      for (let i = 0; i < pageSize; ++i) {
-        const [pageKey] = serialize(getKey(i, previousPageData))
-
-        // Get the cached page data.
-        const pageData = pageKey ? cache.get(pageKey)?.data : UNDEFINED
-
-        // Return the current data if we can't get it from the cache.
-        if (isUndefined(pageData)) return dataRef.current
-
-        data.push(pageData)
-        previousPageData = pageData
-      }
+    const resolvePagesFromCache = useCallback(
+      (pageSize: number, key: string): Data[] | undefined => {
```

Comment:
Q: If we put `infiniteKey` into the deps array of `useCallback`, does the `key` argument get unnecessary?

Reply:
I realize we could just inline this function into `setSize`